examples/node2vec_link.py

```diff
@@ -0,0 +1,147 @@
+import argparse
+import copy
+import os
+
+import torch
+from inferred_stypes import dataset2inferred_stypes
+from text_embedder import GloveTextEmbedding
+from torch.utils.tensorboard import SummaryWriter
+from torch_frame.config.text_embedder import TextEmbedderConfig
+from torch_geometric.nn import Node2Vec
+from torch_geometric.seed import seed_everything
+
+from relbench.data import LinkTask, RelBenchDataset
+from relbench.data.table import Table
+from relbench.data.task_base import TaskType
+from relbench.datasets import get_dataset
+from relbench.external.graph import make_pkey_fkey_graph
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--dataset", type=str, default="rel-trial")
+parser.add_argument("--task", type=str, default="rel-trial-sponsor-condition")
+parser.add_argument("--lr", type=float, default=0.001)
+parser.add_argument("--epochs", type=int, default=3000)
+parser.add_argument("--batch_size", type=int, default=512)
+parser.add_argument("--embedding_dim", type=int, default=128)
+parser.add_argument("--walk_length", type=int, default=10)
+parser.add_argument("--context_size", type=int, default=5)
+parser.add_argument("--num_workers", type=int, default=1)
+parser.add_argument("--log_dir", type=str, default="results")
+args = parser.parse_args()
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+if torch.cuda.is_available():
+    torch.set_num_threads(1)
```

Comment:
why?

Reply:
I was copying this from `gnn_node.py`. Removed here and in the `gnn_node.py` file.

Reply:
actually, this was important to run the script on many datasets and tasks in parallel (otherwise threading causes it to be cpu bottlenecked). Weihua suggested to have this in the public scripts as well, and I have checked that it does not hurt runtime significantly.

Reply:
@yiweny please add back